src/foam/dao/index/AltIndex.java

```diff
@@ -37,13 +37,13 @@ public Object addIndex(Object state, Index i) {
 
     // Copy all data from first index into new index, updating state
     final Object[] sa   = toObjectArray(state);
-          Sink     sink = new AbstractSink() {
+    Sink     sink = new AbstractSink() {
       public void put(Object obj, foam.core.Detachable sub) {
         sa[sa.length-1] = i.put(sa[sa.length-1], (FObject) obj);
       }
     };
 
-    planSelect(sa[0], sink, 0, Long.MAX_VALUE, null, null).select(sa[0], sink, 0, Long.MAX_VALUE, null, null);
+    planSelect(sa, sink, 0, Long.MAX_VALUE, null, null).select(sa[0], sink, 0, Long.MAX_VALUE, null, null);
```

Comment:
Should the second reference to sa[0] also be converted to 'sa'?

Reply:
@kgrgreer it's actually ignored. So more correctly would be to pass null.

Reply:
@kgrgreer planSelect returns an AltSelectPlan which uses it's internal state_ rather than the argument state passed on the select call.